src/TesApi.Web/Controllers/TaskServiceApi.cs

```diff
@@ -159,6 +159,13 @@ public virtual async Task<IActionResult> CreateTaskAsync([FromBody] TesTask tesT
                 ?.Skip(2)
                 ?.FirstOrDefault();
 
+            if (tesTask.Tags == null)
+            {
+                tesTask.Tags = new Dictionary<string, string>();
+            }
+
+            tesTask.Tags["workflow_id"] = tesTask.WorkflowId;
```

Comment:
For the tags we add, we may want to consider using the [set added here](https://github.com/microsoft/tes-azure/blob/11b3b4d2d40f73c4e758c99629ae6dd1b815465d/api-server/tesazure/backends/common/__init__.py#L130C1-L134C105)

Reply:
Started a new thread with the TES folks on what would be canonical, also which casing makes most sense ("workflowId" might be more canonical since JSON is typically camel case).

Reply:
confirmed that TES uses snake case for tag keys to be consistent

Reply:
The thing is that these tags are intended for the customer to populate and we are intruding on their space here. It is accepted internet convention to do the x-organization-other-things-here pattern for naming things in other people's namespaces. We don't want collisions